test/OpenSTUtility.js

```diff
@@ -300,83 +303,86 @@ contract('OpenSTUtility', function(accounts) {
 
             await OpenSTUtility_utils.checkRedemptionIntentDeclaredEvent(result.logs[0], checkBtUuid, checkRedemptionIntentHash, brandedToken,
-			redeemer, 2, 3, unlockHeight, chainIdValue);
+			redeemer, 2, redeemAmountUT, unlockHeight, chainIdValue);
 		})
 	})
 
 	describe('RedeemSTPrime', async () => {
+
+		const redeemSTP = new BigNumber(2);
+
 		before(async () => {
 	        contracts  		 		= await OpenSTUtility_utils.deployOpenSTUtility(artifacts, accounts);
 	        stPrime     			= contracts.stPrime;
 	        openSTUtility 			= contracts.openSTUtility;
 
 	        uuidSTPrime 			= await openSTUtility.uuidSTPrime.call();
-            checkStakingIntentHash 	= await openSTUtility.hashStakingIntent(uuidSTPrime, accounts[0], 1, accounts[0], 5, 5, 80668)
+					checkStakingIntentHash 	= await openSTUtility.hashStakingIntent(uuidSTPrime, accounts[0], 1, accounts[0], amountST, amountUT, 80668)
 
-            await openSTUtility.confirmStakingIntent(uuidSTPrime, accounts[0], 1, accounts[0], 5, 5, 80668, checkStakingIntentHash, { from: registrar });
-            await openSTUtility.processMinting(checkStakingIntentHash);
-			await stPrime.claim(accounts[0]);
+          await openSTUtility.confirmStakingIntent(uuidSTPrime, accounts[0], 1, accounts[0], amountST, amountUT, 80668, checkStakingIntentHash, { from: registrar });
+          await openSTUtility.processMinting(checkStakingIntentHash);
+					await stPrime.claim(accounts[0]);
 	    })
 
 		it('fails to redeem when msg.value is not > 0', async () => {
-            await Utils.expectThrow(openSTUtility.redeemSTPrime(2, { from: redeemer, value: 0 }));
+            await Utils.expectThrow(openSTUtility.redeemSTPrime(redeemSTP.toNumber(), { from: redeemer, value: 0 }));
 		})
 
 		it('fails to redeem when nonce is not >= previously', async () => {
             await Utils.expectThrow(openSTUtility.redeemSTPrime(0, { from: redeemer, value: 2 }));
 		})
 
 		it('successfully redeems', async () => {
-            var redeemReturns = await openSTUtility.redeemSTPrime.call(2, { from: redeemer, value: 2 });
+			var redeemReturns = await openSTUtility.redeemSTPrime.call(redeemSTP.toNumber(), { from: redeemer, value: 2 });
 
-            // call block number is one less than send block number
-            unlockHeight = redeemReturns[1].plus(1)
-            var checkRedemptionIntentHash = await openSTUtility.hashRedemptionIntent.call(uuidSTPrime, redeemer, 2, 2, unlockHeight);
-            result = await openSTUtility.redeemSTPrime(2, { from: redeemer, value: 2 });
+      // call block number is one less than send block number
+      unlockHeight = redeemReturns[1].plus(1)
+      var checkRedemptionIntentHash = await openSTUtility.hashRedemptionIntent.call(uuidSTPrime, redeemer, 2, redeemSTP, unlockHeight);
+      result = await openSTUtility.redeemSTPrime(redeemSTP.toNumber(), { from: redeemer, value: redeemSTP });
```

Comment:
on `redeemSTP.toNumber()` why convert it to a Number? (similarly on L 377, L385)

Reply:
Although it works but you are right it's not required. 
toNumber() is required while comparing values while asserting.